etcdmain/etcd.go

```diff
@@ -379,3 +387,11 @@ func genClusterString(name string, urls types.URLs) string {
 	}
 	return strings.Join(addrs, ",")
 }
+
+// handleInterrupt calls the handler on receiving a SIGTERM.
+func handleInterrupt(h func()) {
+	notifier := make(chan os.Signal)
+	signal.Notify(notifier, os.Interrupt, syscall.SIGTERM)
```

Comment:
os.Interrupt == syscall.SIGINT

But it's probably more confusing than useful. I will change it to syscall.SIGINT for clarity.